packages/node_modules/@ciscospark/redux-module-avatar/src/actions.js

```diff
@@ -57,40 +52,55 @@ export function fetchAvatar({space, userId}, spark) {
       return Promise.resolve();
     }
 
-    dispatch(addAvatarBegin(avatarId));
-
     if (space) {
-      return fetchSpaceAvatar(space, spark, userId)
-        .then(({id, objectUrl}) => {
-          dispatch(addAvatar(id, objectUrl));
-        });
+      return dispatch(fetchSpaceAvatar(space, spark, userId));
     }
-    return fetchUserAvatar(userId, spark)
-      .then(({id, objectUrl}) => {
-        dispatch(addAvatar(id, objectUrl));
-      });
+    return dispatch(fetchUserAvatar(userId, spark));
   };
 }
 
+
 function fetchUserAvatar(userId, spark) {
-  return spark.people.get(userId)
-    .then((person) => Promise.resolve({id: userId, objectUrl: person.avatar}))
-    .catch(() => Promise.resolve({id: userId}));
+  return (dispatch) => {
+    dispatch(addAvatarBegin(userId));
+    return spark.people.get(userId)
+      .then((person) => dispatch(addAvatar(userId, convertToSmallAvatar(person.avatar))))
+      .catch(() => {
+        dispatch(addAvatar(userId));
+        return Promise.resolve({id: userId});
+      });
+  };
 }
 
-function fetchSpaceAvatar(space, spark, userIdForAvatar) {
-  if (space.type === `direct`) {
-    if (!userIdForAvatar) {
-      return Promise.reject(new Error(`Direct spaces require a user id to display`));
+function fetchSpaceAvatar(space, sparkInstance, userIdForAvatar) {
+  return (dispatch) => {
+    if (space.type === `direct` && space.participants.length === 2) {
+      if (!userIdForAvatar) {
+        return Promise.reject(new Error(`Direct spaces require a user id to display`));
+      }
+      return dispatch(fetchUserAvatar(userIdForAvatar, sparkInstance));
     }
-    return fetchUserAvatar(userIdForAvatar, spark);
-  }
-  if (space.avatar) {
-    return spark.internal.conversation.download(space.avatar.files.items[0])
-      .then((file) => {
-        const {objectUrl} = bufferToBlob(file);
-        return Promise.resolve({id: space.id, objectUrl});
-      });
+    if (space.avatar) {
+      dispatch(addAvatarBegin(space.id));
+      return sparkInstance.internal.conversation.download(space.avatar.files.items[0])
+        .then((file) => {
+          const {objectUrl} = bufferToBlob(file);
+          dispatch(addAvatar(space.id, objectUrl));
+          return Promise.resolve({id: space.id});
+        });
+    }
+    return Promise.resolve(dispatch(addAvatar(space.id)));
+  };
+}
+
+/**
+ * A bit of a hack to get smaller Avatars from Public API
+ */
+
+function convertToSmallAvatar(avatarUrl) {
```

Comment:
Awesome, I never got around to putting this in!

Reply:
👍